### DISCUSSION TITLE
Rocket v0.5-rc.1 is Available!

### DISCUSSION THREAD

**Discussion by SergioBenitez** [rwf2/Rocket#1672] [Announcements]
# :rocket: Rocket v0.5-rc.1 is Available! 

Rocket `0.5.0-rc.1`, a release candidate for Rocket v0.5, is now available!

This release actualizes over _two years_ of development, bringing Rocket to Rust stable and `async` to Rocket. Every facet, from core request handling to error messages, has been subjected to Rocket's exacting standards. See the [release announcement for more](https://rocket.rs/v0.5-rc/news/2021-06-09-version-0.5-rc.1/).

A general release is now imminent, but we need your feedback to make it happen. Please take the release candidate for a spin! We encourage all users to upgrade their applications to the release candidate. If you find an issue with the release, please let us know by posting on the [issue tracker](https://github.com/SergioBenitez/Rocket/issues). For any general questions, concerns, or comments, feel free to respond to this discussion or, when appropriate, open a new discussion.

We hope you enjoy Rocket v0.5 as much as we enjoyed creating it!

**Comment by jlshattuck** [comment]
Hi Sergio - I'm getting a 404 on the issue tracker link https://github.com/SergioBenitez/Rocket/issue

**Reply by jlshattuck** [reply]
Looks like it's just missing the trailing "s" in issues

**Reply by SergioBenitez** [reply]
Thanks! Looks like this is just an issue on the discussion post. Fixed the link. :)

**Comment by jplatte** [comment]
What about rocket-contrib? Seems like the 0.5.0-rc.1 version didn't get published on crates.io

**Reply by jplatte** [reply]
Nevermind, was fooled by there still being a `contrib` dir at the repo root, `rocket_dyn_templates` and `rocket_sync_db_pools` have been released 👍🏼

**Reply by jebrosen** [reply]
Features from `rocket_contrib` crate have been [merged into `rocket`], and/or split out into the new crates [`rocket_dyn_templates`] and [`rocket_sync_db_pools`]

[merged into `rocket`]: <https://github.com/SergioBenitez/Rocket/blob/master/CHANGELOG.md#contrib-graduation>
[`rocket_dyn_templates`]: https://api.rocket.rs/v0.5-rc/rocket_dyn_templates/index.html
[`rocket_sync_db_pools`]: https://api.rocket.rs/v0.5-rc/rocket_sync_db_pools/index.html

**Comment by williamdes** [comment]
This is really awesome !! 👏🏻 
I migrated my project that I am starting (folks searching for code examples): https://github.com/air-balloon/api/commit/ed3f4087aa48dea08105f84205e6389424964cf5
And migrated back to Rust stable (Docker multi stage using Alpine): https://github.com/air-balloon/api/commit/f57d66a17fd88ff9df0a443bb6639f8873b8ec1d

Thank you so much for Rust stable support, this will help a lot with simpler configs 👍🏻 

It also feels cleaner on `Cargo.toml`

Other migration example: https://github.com/mountains/permafrost/commit/e7b5476e6ccd87a326c4fcc707c99d855787f4bd

**Comment by JockeTF** [comment]
Awesome! Thanks a ton for this incredible framework!

**Comment by jplatte** [comment]
Two things I noticed while porting [turbo.fish](https://turbo.fish/) that might not be by design / should probably be called out in the migration guide:

* You can no longer have a module and a route handler named the same thing.
* `ROCKET_ADDRESS` no longer allows a value of `localhost` (not sure why I was using it like that, but it used to work, now Rocket fails to start instead)

**Reply by jebrosen** [reply]
> You can no longer have a module and a route handler named the same thing.

Good catch. I assume this is due to structs (generated by codegen) sharing the same namespace as modules, and it definitely deserves a spot in the CHANGELOG/guide.

> ROCKET_ADDRESS no longer allows a value of localhost (not sure why I was using it like that, but it used to work, now Rocket fails to start instead)

Yes, only an actual IP address is now supported. In 0.4 `localhost` was accepted (I assume any hostname that resolved to a local IP also works). This difference should be mentioned in the CHANGELOG, assuming we keep this restriction.

**Comment by imbolc** [comment]
Only a week for testing the RC? Even being on the the master branch, I pulled a couple of moths ago, took me 2 hours to refactor my 8 routes :) Updating `FromRequest` for auth and `FromFormField` for `chrono` types etc. Everything looks better btw :) But I'm afraid people from 0.4 won't have time to migrate and provide their feedback.

**Reply by SergioBenitez** [reply]
I think you raise an excellent concern. Perhaps an RC2 with a full migration guide would give users the support and time to migrate. I'll seriously consider this.

**Reply by williamdes** [reply]
migration guide: https://github.com/SergioBenitez/Rocket/discussions/1672#discussioncomment-1224771

**Comment by TQLeung** [comment]
how can i update the rocket v0.4.7 to v0.5.0-rc1?? have link to guide??

**Reply by blackghost1987** [reply]
From the site: "A complete migration guide and news article will accompany the general release. Until then, the CHANGELOG contains every feature addition, change, and improvement since v0.4."

https://github.com/SergioBenitez/Rocket/blob/v0.5.0-rc.1/CHANGELOG.md

**Reply by memoryruins** [reply]
A migration guide was published a few days ago ^^
https://rocket.rs/master/guide/upgrading-from-0.4/

**Comment by blackghost1987** [comment]
I just wanted to say I managed to upgrade my application to v0.5-rc.1 without any major issues, thanks for all the information in the Changelog, it was really helpful 👍 
Just one comment: I would emphasize the Contrib Graduation a bit more, maybe move it up in the guide, because it was not really clear to me at first how parts of the contrib crate was moved and/or separated into other crates. I've also expected features which have been included in Rocket (like "json") to be usable right away, but it turned out I have to add a feature tag in the Cargo.toml to enable them, that maybe worth a mention too.

Also I know it's a major aspect of v0.5 to be able to build on stable, which is great, but it might be wise to indicate that it's not mandatory. Some of the applications use nightly features anyway (sometimes even without the developer realizing it), or maybe the infrastructure change would be hard. For me it was much easier to first migrate all my stuff to v0.5-rc.1 without changing to stable, and later on change to stable separately.

**Comment by cmayoros** [comment]
Just curious, but this announcement was posted on June 9th. The official announcement on the website says the general release for 0.5 is planned for June 18th. It's August now, and it hasn't been released. What's going on?

**Reply by SergioBenitez** [reply]
It's been an unimaginably difficult month for me, personally, making it difficult to engage outside of these personal matters. Time brings recovery, and so I am certain to make progress on a release imminently.

**Reply by cmayoros** [reply]
I understand. Thanks for the updates. The software can wait, take care of what you need to.

**Reply by xamgore** [reply]
@SergioBenitez is there anything the community could help you with?

**Reply by JLT032** [reply]
I hope things worked out by now, regardless my delayed yet not postponed sympathies.

**Comment by Razican** [comment]
I would love to see compression as part of the Rocket repository. Even if https://github.com/Ameobea/rocket_async_compression exists, having it as part of the usual Rocket distribution would help making it easier to implement.

**Reply by somehowchris** [reply]
Don't know why Ameobea used tokio as a dependency but just forked it and added some stuff https://github.com/somehowchris/rocket_async_compression

**Reply by somehowchris** [reply]
Well got merged back into https://github.com/Ameobea/rocket_async_compression

**Comment by somehowchris** [comment]
@SergioBenitez I hope you are doing fine and I really do not want to be rude, but is the project still on going? There are things left just like they were on the 24th of August and havent been touched, there are PRs & issues open. If I can help you out with something let us/me know

Would really hate to see such an amazing project not being developed and maintained

**Reply by Razican** [reply]
As far as he wrote, there were personal contingencies that prevented him from working in the project. We should give him some slack.

In any case, maybe it would make sense to have more maintainers in the future so that the project doesn't only depend on one person.

**Reply by somehowchris** [reply]
Indeed and as far as I know the last few months before vanishing he got some other people involved but they left. 

Yes I saw his messages but he also wrote a few days ago in another topic that he will post news soon

**Reply by somehowchris** [reply]
Having a repository with over 1k forks and then having to search for a maintained fork is just not the solution to go for but on the other hand projects like Noria get put down as soon as the maintainer "finished" the task which were in his interests and didn't look for maintainers

**Comment by daniel-samson** [comment]
is there a roadmap for the 0.5 release? What is blocking the release?

**Reply by blackghost1987** [reply]
The only crate maintainer (Sergio) is not available for some time now for personal reasons, and we don't know where will he be back. The earliest he might be back is in the summer, so an 0.5 release is unlikely for a while.
We are trying to move the crate to a new GitHub organization (https://github.com/rocket-org/) to give more people release rights, see the discussion here: https://github.com/SergioBenitez/Rocket/discussions/2011

It's possible the organization can orchestrate a 0.5 release sooner, but if Sergio won't explicitly give release rights to the Org, then the release will have to be a new crate in Crates.io unfortunately

**Reply by daniel-samson** [reply]
I hope that it doesn't come to that. Rocket is awesome!!! I hope @SergioBenitez that you can come back or look for a maintainer to help out.
